tests/test_metric_output.py

```diff
@@ -0,0 +1,39 @@
+import requests
+import yaml
+
+
+class TestMetricOutput:
+    def test_vcsa_data_comparison(self, setup_vcsa_url):
+        response = requests.get(setup_vcsa_url)
+        metrics = self.process_response_data(response)
+
+        with open('tests/vcsa_output_comparison.yaml', 'r') as fp:
+            comparison_data = yaml.safe_load(fp)
+
+        for key in comparison_data.keys():
+            difference = set(comparison_data[key]).difference(metrics)
+            assert len(difference) == 0
```

Comment:
You are expecting a difference here, which makes sense from what you are doing earlier. Although, what is the purpose of this test? A proper name would be `test_prometheus_server_reply` or comparable. You are just comparing the plain output of prometheus with an always divergent yaml.

Reply:
I am not expecting a difference at this part. I compare the vcsa metrics with a static metric collection that I dumped from the vcsa at a working state.  If the output from the exporter differs from the expected yaml file (len(difference) > 0), the testcase fails. It is the main exporter test, which tests if the exporter as a whole is delivering the correct output.

Reply:
done.